### PR TITLE
Update "New for 2021" article

### PR DESCRIPTION
**Pull request by prateekma**
Adds some more changes between Beta 1 release day and present-day.

### CODE REVIEW

File: source/docs/yearly-overview/yearly-changelog.rst [hunk 1]
```diff
@@ -23,6 +24,8 @@ Breaking Changes
 
 - Trajectory constraint methods are now ``const`` in C++. Teams defining their own custom constraints should mark the ``MaxVelocity()`` and ``MinMaxAcceleration()`` methods as ``const``.
 
+- The ``Field2d`` class was moved from the simulation package (``edu.wpi.first.wpilibj.simulation`` / ``frc/simulation/``) to the SmartDashboard package (``edu.wpi.first.wpilibj.smartdashboard`` / ``frc/SmartDashboard/``). This allows teams to send their robot position over NetworkTables to be viewed in Glass.
```
Comment (sciencewhiz) [inline source/docs/yearly-overview/yearly-changelog.rst:27]: This should tell them to put it to smartdashboard in order to see it in the simgui.

Comment (sciencewhiz) [reply source/docs/yearly-overview/yearly-changelog.rst]: I think it's still not clear that you have to put it to SD in order to have the old behavior of it showing in the simgui

**Comment by Daltz333** [comment]
I feel like there was a couple other things in some of the other repos that was merged?

**Comment by prateekma** [comment]
@Daltz333 I'm not aware of any noteworthy changes we have made to the other repos (i.e. tools or frc-characterization) that I'm comfortable with explaining.

**Comment by jasondaming** [comment]
maybe we leave a note that frc-characterization data or config files made in the past no longer work with the current frc-characterization?

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**